app/models/user.rb

```diff
@@ -1,4 +1,25 @@
 class User < ActiveRecord::Base
   has_many :bids
-  validates :email, email: true, allow_blank: true
+
+  validates :email, allow_blank: true, email: true
+  validate :validate_credit_card_form
```

Comment:
did you consider using a lib for this? https://github.com/perfectline/validates_url

That way, this could just be a one-liner. Might feel kind of heavy-handed to introduce a lib for this small thing (and I'd check on whether the lib itself is high quality first) but my guess is that, like with email addresses, there are a good number of edge cases here that our little method below might not cover. This is why I always use a lib for validation format of email addresses. Let someone else deal with those edge cases (in a well-tested way)!